src/Microsoft.AspNetCore.Rewrite/RewriteOptionsExtensions.cs

```diff
@@ -12,9 +13,36 @@ namespace Microsoft.AspNetCore.Rewrite
     public static class RewriteOptionsExtensions
     {
         /// <summary>
+        /// Set the default path base used to find config files
+        /// </summary>
+        /// <remarks>
+        /// This should be called before using
+        /// <see cref="ApacheModRewriteOptionsExtensions.AddApacheModRewrite(RewriteOptions, string)"/> or
+        /// <see cref="IISUrlRewriteOptionsExtensions.AddIISUrlRewrite(RewriteOptions, string)"/>.
+        /// </remarks>
+        /// <param name="options">The <see cref="RewriteOptions"/></param>
+        /// <param name="pathBase">The root path to use</param>
+        /// <returns></returns>
+        public static RewriteOptions SetConfigFileBasePath(this RewriteOptions options, string pathBase)
```

Comment:
would it make more sense to have this take IHostinEnvironment or IFileProvider? Config doesn't do that because they don't exist at that stage, but they will exist when configuring middleware.